src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +102,1361 @@ static DamageTable D_80BBF858 = {
+        }
+    } else if (curFrame == lastFrame) {
+        this->actionFlags &= ~ENSGORO_ACTIONFLAG_TIRED;
+        this->animInfoIndex = ENSGORO_ANIM_IDLE_STAND;
+        SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+    }
+}
+
+void EnSGoro_UpdateToIdleAnimation(EnSGoro* this) {
+
+    s16 curFrame = this->skelAnime.curFrame;
+    s16 lastFrame = Animation_GetLastFrame(sAnimationInfo[this->animInfoIndex].animation);
+
+    if ((this->animInfoIndex != ENSGORO_ANIM_IDLE_STAND) && (curFrame == lastFrame)) {
+        this->animInfoIndex = ENSGORO_ANIM_IDLE_STAND;
+        SubS_ChangeAnimationByInfoS(&this->skelAnime, sAnimationInfo, this->animInfoIndex);
+    }
+}
+
+void EnSGoro_UpdateCollider(EnSGoro* this, PlayState* play) {
+
+    Vec3f world_pos = this->actor.world.pos;
+    f32 radius = 24.0f;
+    f32 height = 62.0f;
+
+    this->collider.dim.pos.x = world_pos.x;
+    this->collider.dim.pos.y = world_pos.y;
+    this->collider.dim.pos.z = world_pos.z;
+    this->collider.dim.radius = radius;
+    this->collider.dim.height = height;
+
+    if ((s32)this != -0x190) {
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    }
+}
+
+void EnSGoro_UpdateEyes(EnSGoro* this) {
+
+    s16 eyeCtr;
+
+    if ((this->actionFlags & ENSGORO_ACTIONFLAG_EYESOPEN)) {
+        if (this->eyeTimer == 0) {
+            eyeCtr = 0;
+        } else {
+            this->eyeTimer--;
+            eyeCtr = this->eyeTimer;
+        }
+
+        if (eyeCtr == 0) {
+            this->eyeTexIndex++;
+            if (this->eyeTexIndex >= EYETEX_INDEX_MAX) {
+                this->eyeTimer = Rand_S16Offset(30, 30);
+                this->eyeTexIndex = ENSGORO_EYETEX_OPEN;
+            }
+        }
+    }
+}
+
+void EnSGoro_UpdateActorFocus(EnSGoro* this) {
+
+    f32 yDelta;
+
+    if ((this->actionFlags & ENSGORO_ACTIONFLAG_ROLLEDUP)) {
+        yDelta = this->actor.shape.yOffset;
+    } else {
+        yDelta = 58.0f;
+    }
+
+    this->actor.focus.pos.x = this->actor.world.pos.x;
+    this->actor.focus.pos.y = this->actor.world.pos.y + yDelta;
+    this->actor.focus.pos.z = this->actor.world.pos.z;
+    this->actor.focus.rot.x = this->actor.world.rot.x;
+    this->actor.focus.rot.y = this->actor.world.rot.y;
+    this->actor.focus.rot.z = this->actor.world.rot.z;
+}
+
+s32 EnSGoro_UpdateRotationToPlayer(EnSGoro* this, PlayState* play) {
+
+    Player* player = GET_PLAYER(play);
+    Vec3f playerPos;
+    Vec3f thisPos;
+    s16 target = (this->actor.yawTowardsPlayer - this->bodyRotY) - this->actor.shape.rot.y;
+    s32 pval;
+
+    Math_SmoothStepToS(&this->headRotY, target, 4, 0x2AA8, 1);
+    if (this->headRotY < -0x1FFE) {
+        this->headRotY = -0x1FFE;
+    } else {
+        if (this->headRotY >= 0x1FFF) {
+            pval = 0x1FFE;
+        } else {
+            pval = this->headRotY;
+        }
+        this->headRotY = pval;
```

Comment:
```suggestion
    CLAMP(this->headRotY, -0x1FFE, 0x1FFE);
```
Also looks like a `CLAMP`